### PR TITLE
Merging neural network code and repository structure

### PR DESCRIPTION
**Pull request by jackmiller2003**
This merge contains the relevant code for the neural network experiments. Explanation of setup etc. is available in the new README.

### CODE REVIEW

**Comment by jackmiller2003** [comment]
@vboussange we should merge this into main so that Youssef can do the data analysis.

File: analysis/lorenz/lorenz_with_all_seeds.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import json
```
Comment (vboussange) [inline analysis/lorenz/lorenz_with_all_seeds.py:1]: Missing preamble

File: analysis/sst/sst_with_hessian_plots.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
+"""
+Script for generating relevant plots associated with training a neural network on the Lorenz system.
```
Comment (vboussange) [inline analysis/sst/sst_with_hessian_plots.py:2]: I guess you mean SST here

File: analysis/sst/train_all_with_sst.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
+import json
```
Comment (vboussange) [inline analysis/sst/train_all_with_sst.py:1]: Missing preamble

File: experiments/climsim/train_mlp_on_climsim_minimal.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
+"""
+Example experiment demonstrating iterative increase of prediction horizon compared to training
+with given horizons directly.
```
Comment (vboussange) [inline experiments/climsim/train_mlp_on_climsim_minimal.py:3]: This preamble seems erroneous. Here you simply train MLP with fixed temporal horizon, right?

File: experiments/lorenz/calculating_hessian_on_lorenz.py
Comment (vboussange) [inline experiments/lorenz/calculating_hessian_on_lorenz.py:1]: Can you better explain the difference in all scripts in experiments/Lorenz?

File: src/hessians.py
Comment (vboussange) [inline src/hessians.py:1]: Can you better document those classes?

File: src/models/gnn.py
Comment (vboussange) [inline src/models/gnn.py:1]: Where did you find this architecture?

**Review by vboussange — COMMENTED** [review]
Nice work! Let's discuss it at our meeting

**Pull request merged**